code/modules/mob/living/simple_animal/abnormality/he/funeral.dm

```diff
@@ -0,0 +1,258 @@
+						ABNORMALITY_WORK_REPRESSION = list(0, 0, 60, 60, 60),
+						)
+	work_damage_amount = 12
+	work_damage_type = WHITE_DAMAGE
+	deathmessage = "floats into the air, descending into its own coffin."
+
+	ego_list = list(
+		/datum/ego_datum/weapon/solemnvow,
+		/datum/ego_datum/weapon/solemnlament,
+		/datum/ego_datum/armor/solemnlament
+		)
+	gift_type =  /datum/ego_gifts/solemnlament
+	var/gun_cooldown
+	var/gun_cooldown_time = 4 SECONDS
+	var/gun_damage = 60
+	var/swarm_cooldown
+	var/swarm_cooldown_time = 30 SECONDS
+	var/swarm_damage = 12
+	var/can_act = TRUE
+/datum/action/innate/abnormality_attack/SpiritGun
+	name = "Bring to Rest"
+	icon_icon = 'icons/obj/wizard.dmi'
+	button_icon_state = "scroll"
+	chosen_message = "<span class='colossus'>You will now fire butterflies.</span>"
+	chosen_attack_num = 1
+
+/datum/action/innate/abnormality_attack/ButterflySwarm
+	name = "Empty Casket"
+	icon_icon = 'icons/obj/wizard.dmi'
+	button_icon_state = "scroll"
+	chosen_message = "<span class='colossus'>You will now unleash a swarm of butterflies.</span>"
+	chosen_attack_num = 2
+/mob/living/simple_animal/hostile/abnormality/funeral/AttackingTarget(atom/attacked_target)
+	return OpenFire()
+
+/mob/living/simple_animal/hostile/abnormality/funeral/OpenFire()
+	if(!can_act)
+		return
+	if(client)
+		switch(chosen_attack)
+			if(1)
+				SpiritGun(target)
+			if(2)
+				ButterflySwarm(target)
+		return
+	if(swarm_cooldown <= world.time && prob(75))
+		ButterflySwarm(target)
+	else
+		SpiritGun(target)
+	return
+
+
+/mob/living/simple_animal/hostile/abnormality/funeral/proc/SpiritGun(atom/target)
+	if(!istype(target, /mob/living)||gun_cooldown > world.time)
+		return
+	var/mob/living/cooler_target = target
+	can_act = FALSE
+	visible_message("<span class='danger'>[src] levels one of its arms at [cooler_target]!</span>")
+	gun_cooldown = world.time + gun_cooldown_time
+	SLEEP_CHECK_DEATH(1 SECONDS)
+	playsound(get_turf(src), 'sound/abnormalities/funeral/spiritgun.ogg', 75, 1, 3)
+	if(cooler_target in oview(src, 12))
+		cooler_target.apply_damage(60, WHITE_DAMAGE, null, cooler_target.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
+		//No longer because fuck you.
+		if(istype(target, /mob/living/carbon/human))
+			var/mob/living/carbon/human/kickass_grade1_target = target
+			//I'm not sorry.
+			if(kickass_grade1_target.sanity_lost)
+				kickass_grade1_target.apply_damage(9999, PALE_DAMAGE, null, kickass_grade1_target.run_armor_check(null, PALE_DAMAGE), spread_damage = TRUE)
+	//This should be brute. But also, it's funny.
+	can_act = TRUE
+
+
+
+/mob/living/simple_animal/hostile/abnormality/funeral/proc/ButterflySwarm(target)
+	if(swarm_cooldown > world.time)
+		return
+	if (get_dist(src, target) < 5)
+		return
+	var/dir_to_target = get_cardinal_dir(get_turf(src), get_turf(target))
+	var/turf/source_turf = get_turf(src)
+	var/turf/area_of_effect = list()
+	var/turf/upper_bound
+	var/turf/lower_bound
+	switch(dir_to_target)
```

Comment:
I'll be honest, copying woodsman here is a bad idea; Will probably need a rework.